### PR TITLE
Add support for binding to IPv6 localhost in `flutter run`

### PR DESCRIPTION
**Pull request by tvolkert**
This allows `flutter run` to work on hosts that are IPv6-only.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -28,6 +28,12 @@ abstract class RunCommandBase extends FlutterCommand {
         negatable: true,
         defaultsTo: false,
         help: 'Start tracing during startup.');
+    argParser.addFlag('ipv6',
+        hide: true,
+        negatable: false,
+        defaultsTo: false,
+        help: 'Binds to IPv6 localhost instead of IPv4 when Flutter Tools\n'
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: I believe we usually refer to this as "the flutter tool" or some such.

**Comment by Hixie** [comment]
LGTM

**Review by jakobr-google — COMMENTED** [review]
DBC. There are a couple of related IPv6 issues still, I think:

* On a pure IPv6 host, HostPortScanner.findPreferredPort() will never return the preferred port. It'll always return a random available port, since isPortAvailable() always returns false (because it fails to bind to the port on IPv4 loopback).
* In a dual-stack system, the PortForwarder will (most likely) bind to a port on IPv4 loopback, but we're now returning a URI pointing to the same port on IPv6 loopback. I *believe* at least Linux will bind the port to both IPv4 and IPv6 loopback, so this probably works, but are we sure? This applies even if we don't use a forwarder, since we unconditionally replace the host with IPv6 loopback if `ipv6` is true, regardless of which stack the port is bound to.

**Comment by tvolkert** [comment]
@jakobr-google good points.  This PR unblocked the Google internal use case, since it uses a dedicated `PortScanner` to obtain leases on ports in a shared machine, then we pass that port to flutter tools explicitly.  And since the machines are Linux and do bind the port to both IPv4 and IPv6 loopback, that works too.  But yes, there's more to do before we declare this 100% IPv6 compliant.

**Pull request merged**